### ISSUE TITLE
How do you hide root from apps inside Shelter?

### ISSUE THREAD

**Issue by fummymint** [PeterCxy/Shelter#130]
I put the Nintendo app inside the Shelter workspace but it does not work because it detects root and I cannot see the option in Magisk to hide the Nintendo app. And no hiding Magisk systemwide does not work, already tried it. 

It works fine outside Shelter because I can hide root. Does anyone have a workaround this?

**Comment by xcffl** [comment]
Clone Magisk manager to work profile

**Comment by fummymint** [comment]
> Try cloning Magisk manager to work profile

Would cloning system apps like magisk even work in work profile?

**Comment by PeterCxy** [comment]
Clone Magisk to work profile, then choose "User-Independent" under "Multiuser Mode" in the settings of the Magisk Manager OUTSIDE the work profile. After doing this, the Magisk Manager inside Work will work just like the one outside and you can enable MagiskHide.

**Comment by xcffl** [comment]
Yeah, the only problem of magisk manager is that it doesn't iteratively check the app list of all users. As a system app with root access, in managed profiles it can also inherit the priviledge (namespace, probably) of its sister in main user.

**Comment by bumjrah** [comment]
Can confirm that this is now working on Magisk v 21.1
